src/config.js

```diff
@@ -104,9 +106,14 @@ function getExecutableParams(params = {}) {
   }
   if (proxy && proxy !== '') {
     const proxyAgent = new HttpsProxyAgent(proxy);
-    config.httpOptions = {
-      httpRequestOptions: { agent: proxyAgent },
-      httpsRequestOptions: { agent: proxyAgent },
+    config.httpOptions.httpRequestOptions = { agent: proxyAgent };
+    config.httpOptions.httpsRequestOptions = { agent: proxyAgent };
+  }
+
+  if (finalUrl.username !== '' || finalUrl.password !== '') {
```

Comment:
What would happen if one of them is missing? Should the two of them be required?

Reply:
👍